### PR TITLE
Added hash interface

### PR DESCRIPTION
**Pull request by mikelodder7**
Added hash interface and a few misc fixes

### CODE REVIEW

File: libhl-crypto/Cargo.toml [hunk 1]
```diff
@@ -42,6 +42,7 @@ sha2 = "0.7.1"
 sha3 = "0.7.3"
 time = "0.1.36"
 env_logger = "0.4.3"
+blake2b_simd = "0.3.1"
```
Comment (dcmiddle) [inline libhl-crypto/Cargo.toml:45]: This library looks pretty slick but I think we need to establish some criteria for new libraries. Probably too extensive for PR comment so I'll take it to the list. Initial reservations here though include that it has a single developer and only a few months of activity.

**Review by dcmiddle — CHANGES_REQUESTED** [review]

File: libhl-crypto/src/hash/blake2.rs [hunk 2]
```diff
@@ -37,102 +37,3 @@ impl_hasher!(Blake2b384, 48);
...
-//}
-//
-//#[cfg(test)]
-//mod tests {
-//
-//    #[test]
-//    fn test_vectors() {
-//
-//    }
-//}
```
Comment (dcmiddle) [inline libhl-crypto/src/hash/blake2.rs]: Probably better if you can squish this into the commit that introduces the file.

**Comment by mikelodder7** [comment]
It’s the library that the blake2 authors support

**Review by vaporos — APPROVED** [review]

File: libhl-crypto/Cargo.toml (hunk 1 above)
Comment (mikelodder7) [reply libhl-crypto/Cargo.toml:45]: It’s the library that the blake2 authors support (https://blake2.net) see [here](https://github.com/oconnor663/blake2_c.rs)

Comment (dcmiddle) [reply libhl-crypto/Cargo.toml:45]: Per lab discussion the full definition of tiers and review criteria will be formalized as the project launches. Meanwhile no reason to hold up this PR. We will likely have to refactor this blake2 implementation (or any other) into the right tier once we define what 'tier' means in terms of code, namespace, etc.

**Review by dcmiddle — APPROVED** [review]

**Pull request merged**